californium-core/src/main/java/org/eclipse/californium/core/network/InMemoryMessageExchangeStore.java

```diff
@@ -61,21 +61,28 @@
 	private ScheduledFuture<?> statusLogger;
 	private ScheduledExecutorService scheduler;
 	private MessageIdProvider messageIdProvider;
+	private TokenProvider tokenProvider;
 	private SecureRandom secureRandom;
 
 	/**
 	 * Creates a new store for configuration values.
 	 * 
 	 * @param config the configuration to use.
+	 * @param tokenProvider the TokenProvider which provides CoAP tokens that
+	 *            are guaranteed to be not in use.
+	 * 
 	 */
-	public InMemoryMessageExchangeStore(final NetworkConfig config) {
+	public InMemoryMessageExchangeStore(final NetworkConfig config, TokenProvider tokenProvider) {
 		if (config == null) {
 			throw new NullPointerException("Configuration must not be null");
-		}else {
-			this.config = config;
-			this.tokenLength = config.getInt(NetworkConfig.Keys.TOKEN_SIZE_LIMIT, MAX_TOKEN_LENGTH);
-			LOGGER.log(Level.CONFIG, "using tokens of {0} bytes in length", tokenLength);
 		}
+		if (tokenProvider == null) {
+			throw new NullPointerException("TokenProvider must not be null");
+		}
+		this.tokenProvider = tokenProvider;
+		this.config = config;
+		this.tokenLength = config.getInt(NetworkConfig.Keys.TOKEN_SIZE_LIMIT, MAX_TOKEN_LENGTH);
```

Comment:
Do we still need the token length here? It seems that the `TokenProvider` retrieves it from the config itself, doesn't it?